spec/controllers/idv/review_controller_spec.rb

```diff
@@ -0,0 +1,148 @@
+require 'rails_helper'
+
+describe Idv::ReviewController do
+  render_views
+
+  let(:user) { create(:user, :signed_up, email: 'old_email@example.com') }
+  let(:user_attrs) do
+    {
+      first_name: 'Some',
+      last_name: 'One',
+      ssn: '666661234',
+      dob: 'March 29, 1972',
+      address1: '123 Main St',
+      address2: '',
+      city: 'Somewhere',
+      state: 'KS',
+      zipcode: '66044',
+      phone: user.phone,
+      ccn: '12345678'
+    }
+  end
+
+  describe 'before_actions' do
+    it 'includes before_actions from AccountStateChecker' do
+      expect(subject).to have_actions(
+        :before,
+        :confirm_two_factor_authenticated,
+        :confirm_idv_session_started
+      )
+    end
+  end
+
+  describe '#new' do
+    before do
+      sign_in(user)
+      allow(subject).to receive(:confirm_two_factor_authenticated).and_return(true)
+      allow(subject).to receive(:confirm_idv_session_started).and_return(true)
+      allow(subject).to receive(:current_user).and_return(user)
+    end
+
+    context 'user has completed all steps' do
+      before do
+        allow(subject).to receive(:idv_session).and_return(params: user_attrs)
+      end
+
+      it 'shows completed session' do
+        get :new
+
+        user_attrs.values.each do |attr|
+          expect(response.body).to have_content(attr)
```

Comment:
A view spec would be better to test this, as it is faster than a controller spec that renders views. Since the `new` action in this controller only symbolizes the keys in a hash, I would write a single test that makes sure the keys in the hash get symbolized. That is all the action is doing.

I would then write separate specs for the before_action methods to unit test them since we already have a spec that makes sure this controller includes those before_actions.